src/AbstractEnum.php

```diff
@@ -2,24 +2,30 @@
 
 namespace Greg0ire\Enum;
 
+use Doctrine\Common\Inflector\Inflector;
+
 /**
  * @author Grégoire Paris <postmaster@greg0ire.fr>
  * @author Sullivan Senechal <soullivaneuh@gmail.com>
  */
 abstract class AbstractEnum
 {
+    const DEFAULT_NAMESPACE_SEPARATOR = '_';
+
     private static $constCache = [];
 
     /**
      * Uses reflection to find the constants defined in the class and cache
      * them in a local property for performance, before returning them.
      *
      * @param callable|null $keysCallback
+     * @param bool          $classPrefixed      True if you want the enum class prefix on each keys, false otherwise.
+     * @param string        $namespaceSeparator Only relevant if $classPrefixed is set to true.
      *
      * @return array a hash with your constants and their value. Useful for
      *               building a choice widget
      */
-    final public static function getConstants($keysCallback = null)
+    final public static function getConstants($keysCallback = null, $classPrefixed = false, $namespaceSeparator = self::DEFAULT_NAMESPACE_SEPARATOR)
```

Comment:
So you are creating your own convention.

![image](https://cloud.githubusercontent.com/assets/1698357/16269245/cf0592d2-3891-11e6-9f89-1a636d09d846.png)

But right, will do it.

![right](https://camo.githubusercontent.com/c5ae9712cb05a228bdb8fa469b5fb1457ef977a5/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f313042314f4b56755a423553726d2f67697068792e676966)